### PR TITLE
test: 96% coverage of split expense

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=WhCFdql)
### <samp>🤖 Generated by Copilot at f4630f6</samp>

This pull request enhances the unit testing of the `split-expense.page.ts` component by adding and updating import statements, variables, and test cases in the `split-expense.page.spec.ts` file. It also makes two methods public in the component class, and adds mock data and model files for the CurrencyObj and MatchedCCCTransaction interfaces.

### <samp>🤖 Generated by Copilot at f4630f6</samp>

> _We split the expense of doom_
> _With mock data and interface_
> _We test the methods of our code_
> _With public access and import case_

### <samp>🤖 Generated by Copilot at f4630f6</samp>

*  Add mock data files for currency objects and matched CCC transactions ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-5ea12bdb0fc7d819953dd88adee362a5f47791e5156244f58ba7d3e502e5729aR1-R8), [link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-13948121cb04c2883782c67a06db3f0a347453de0ad6a33853ed25fe80619b43R1-R21))
*  Add model file for currency objects ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-50ed6f7e6b117e54776e5c13b548047851cf227684bc1956d5c2381c9e3fd98cR1-R6))
*  Update import statements in split expense page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5L39-R45), [link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R82), [link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R131-R134))
*  Set up initial state of component in beforeEach block of split expense page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R1017-R1018))
*  Add test cases for ionViewWillEnter method of split expense page class ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R1078-R1148))
*  Add test cases for save method of split expense page class ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R1614-R1741))
*  Remove private modifiers from setEvenSplit and isEvenlySplit methods of split expense page class ([link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L688-R688), [link](https://github.com/fylein/fyle-mobile-app/pull/2241/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L706-R706))

## Clickup
https://app.clickup.com/t/85ztjmt82

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Comment by suyashpatil78** [comment]
Will fix the lint issues later before merging into master

File: src/app/core/mock-data/matchedCCCTransaction.data.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  currency: 'USD',
+  description: null,
+  group_id: 'ccceP7Kl5Vk6a5',
+  id: 'ccceP7Kl5Vk6a5',
+  orig_amount: null,
+  orig_currency: null,
+  payment_id: 'payx52GN91Sgy',
+  settlement_id: 'setz25bnskJLD',
+  state: 'IN_PROGRESS',
+  txn_dt: new Date('2019-07-13T00:00:00.000Z'),
```
Comment (Dimple16) [inline src/app/core/mock-data/matchedCCCTransaction.data.ts:18]: is it coming in date format in the API response? verify this once

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 2]
```diff
@@ -1619,4 +1623,132 @@ describe('SplitExpensePage', () => {
       expect(isEvenSplit).toBe(true);
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1623]: `toBeTrue();`

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 3]
```diff
@@ -1619,4 +1623,132 @@ describe('SplitExpensePage', () => {
...
+
+      component.save();
+
+      expect(component.showErrorBlock).toBeTrue();
+      expect(component.errorMessage).toEqual('Split amount cannot be more than 2000.');
+      tick(2500);
+      expect(component.showErrorBlock).toBeFalse();
+    }));
+
+    it('should return void and show error message if expense is less than 0.01', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1664]: `should return void` can skip this, ideally we aren't returning anything, use:
`should show an error message if the expense amount is less than 0.01 and return`

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 4]
```diff
@@ -1619,4 +1623,132 @@ describe('SplitExpensePage', () => {
...
+        percentage: [50],
+        txn_dt: [new Date(), Validators.compose([Validators.required, component.customDateValidator])],
+      });
+      component.splitExpensesFormArray = new FormArray([mockSplitExpForm]);
+
+      component.save();
+
+      expect(component.showErrorBlock).toBeTrue();
+      expect(component.errorMessage).toEqual('Amount should be greater than 0.01');
+      tick(2500);
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1680]: do we need `2500`?

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 5]
```diff
@@ -1619,4 +1623,132 @@ describe('SplitExpensePage', () => {
...
+      expect(component.showErrorBlock).toBeFalse();
+    }));
+
+    it('should return void and show error message if expense is less than 0.01', () => {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1684]: this test case has the same description, this is incorrect no?

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 6]
```diff
@@ -684,7 +684,7 @@ export class SplitExpensePage implements OnInit {
     this.getTotalSplitAmount();
   }
 
-  private setEvenSplit(evenAmount, evenPercentage, lastSplitAmount, lastSplitPercentage) {
+  setEvenSplit(evenAmount, evenPercentage, lastSplitAmount, lastSplitPercentage) {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.ts:687]: why remove these as private? 
add a `// @ts-ignore` in the test case instead?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 5 above)
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:1684]: Yes, changing the test description!

File: src/app/core/mock-data/matchedCCCTransaction.data.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/core/mock-data/matchedCCCTransaction.data.ts]: It is coming in the form of `string`, changing the models and respective mock data

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 4 above)
Comment (OmkarJ13) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:1681]: @suyashpatil78 Can you pls add a code comment mentioning this ^

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 4 above)
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:1691]: @OmkarJ13 done!

**Pull request merged**